rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -188,7 +197,22 @@ class Publisher : public PublisherBase
   virtual void
   publish(std::unique_ptr<MessageT, MessageDeleter> & msg)
   {
-    this->do_inter_process_publish(msg.get());
+    size_t inter_process_subscritpion_count;
+    size_t intra_process_subscritpion_count = 0;
+
+    inter_process_subscritpion_count = this->get_subscription_count();
```

Comment:
@ivanpauno `inter_process_subscritpion_count` -> `inter_process_subscription_count`.

Reply:
Also, consider declaring and initializing in the same statement.